### PR TITLE
Issue 344: Rename enw_delay_filter()

### LINKED ISSUE #344: enw_filter_delay function: Filter based on report - reference date
**Is your feature request related to a problem? Please describe.**
When using `epinowcast` it is often useful to compare nowcasts against later observed data. When doing this there are two sensible choices. The first is supported and is to filter the latest data at some future point and then use `enw_latest_data()`.  The second is to filter the data a consistent delay after the reference date so that all reference dates have the same horizon. This is currently unsupported. 

**Describe the solution you'd like**
A new function called `enw_filter_delay` or similar that is based on the current `enw_filter_reference` and `enw_filter_report` functions and enables users to filter out reported data greater than some maximum delay. 

The new function should be clearly documented and have an example workflow. It should also have an informative news item that lays out why a user might want to use it.

### PR DESCRIPTION
**Pull request by kathsherratt**
## Description

This PR closes #344.

- Renames enw_delay_filter() to enw_filter_delay()
- Deprecated enw_delay_filter() with hard stop error (though this might be too harsh... I later noticed previous epinowcast depreciations used deprecate_warn() )

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [x] I have updated the package development version by one increment in both `NEWS.md` and the `DESCRIPTION`.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -51,6 +51,9 @@ This release is in development and not yet ready for production use.
 - Updated the README to include the non-parametric reference date model as an option and also added a new example showing how to use this model. See #313 by @seabbs and self-reviewed.
 - Added a new example showcasing how to fit a model to data reported weekly with a 3 day delay until any reports are non-zero with a weekly process model and a mixture of a parametric and non-parametric reference date model. See #348 by @seabbs and self-reviewed.
 
+## Depreciation
+- `enw_delay_filter()`: Deprecated with an error in favour of `enw_filter_delay()`. This renaming is to better reflect the function's purpose: see #344.
```
Comment (seabbs) [inline NEWS.md:55]: can you add yourself as the PR author and me as the review (@seabbs).

Comment (seabbs) [reply NEWS.md:55]: Potentially the title should be Depreciations?

File: R/preprocess.R [hunk 2]
```diff
@@ -492,6 +492,27 @@ enw_latest_data <- function(obs) {
...
+#' @description `r lifecycle::badge("deprecated")`
+#' @return A `data.frame` filtered so that dates by report are less than or
+#' equal the reference date plus the maximum delay.
+#'
+#' @inheritParams enw_add_incidence
+#' @inheritParams enw_preprocess_data
+#' @importFrom lifecycle deprecate_stop
+#' @family preprocess
+#' @export
+enw_filter_delay <- function(obs, max_delay, timestep = "day") {
```
Comment (seabbs) [inline R/preprocess.R:506]: I think we might want to stick with using `deprecate_warn` for now. Unless you feel strongly?

File: R/preprocess.R [hunk 3]
```diff
@@ -492,6 +492,27 @@ enw_latest_data <- function(obs) {
...
+  lifecycle::deprecate_stop(
+    when = "0.2.3",
+    what = "enw_delay_filter()",
+    with = "enw_filter_delay()",
+    details = "Please file an issue if deprecating this \
+      function has caused any issues."
+  )
+}
+
 #' Filter observations to restrict the maximum reporting delay
```
Comment (seabbs) [inline R/preprocess.R:516]: Could you have a crack at rewording this to better fit what the function does? It currently doesn't really match the name when I read it?

**Review by seabbs — CHANGES_REQUESTED** [review]
Thanks @kathsherratt! This is great. I think we probably want to stick with deprecate_warn here unless you think strongly? 

Once you've had a chance to think about my comments could you ping me for another review?

Could you also add yourself to the package DESCRIPTION as a contributor?

File: R/preprocess.R (hunk 3 above)
Comment (kathsherratt) [reply R/preprocess.R]: Changed it a bit but not sure I got this right tbh

**Comment by kathsherratt** [comment]
@seabbs pinging for a second review

**Review by seabbs — APPROVED** [review]
LGTM. Thanks a lot Kath!

File: NEWS.md [hunk 4]
```diff
@@ -51,6 +51,9 @@ This release is in development and not yet ready for production use.
 - Updated the README to include the non-parametric reference date model as an option and also added a new example showing how to use this model. See #313 by @seabbs and self-reviewed.
 - Added a new example showcasing how to fit a model to data reported weekly with a 3 day delay until any reports are non-zero with a weekly process model and a mixture of a parametric and non-parametric reference date model. See #348 by @seabbs and self-reviewed.
 
+## Depreciations
+- `enw_delay_filter()`: Deprecated with an error in favour of `enw_filter_delay()`. This renaming is to better reflect the function's purpose. See #365 by @kathsherratt and reviewed by @seabbs.
```
Comment (jamesmbaazam) [inline NEWS.md:55]: I believe this was changed to warn instead of error.

**Review by jamesmbaazam — COMMENTED** [review]
Just noticed a minor mistake in the news item.

File: NEWS.md (hunk 4 above)
Comment (seabbs) [reply NEWS.md:55]: yes that is correct - good catch @jamesmbaazam! I guess we need a new issue and PR volunteer 🙈

Comment (jamesmbaazam) [reply NEWS.md:55]: Sure. 😅

**Pull request merged**